src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -0,0 +1,152 @@
+import com.google.api.codegen.transformer.SurfaceTransformerContext;
+import com.google.api.codegen.util.java.JavaTypeTable;
+import com.google.api.codegen.viewmodel.InitCodeView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.codegen.viewmodel.testing.GapicSurfaceTestAssertView;
+import com.google.api.codegen.viewmodel.testing.GapicSurfaceTestCaseView;
+import com.google.api.codegen.viewmodel.testing.GapicSurfaceTestClassView;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import java.util.ArrayList;
+import java.util.List;
+
+/** A subclass of ModelToViewTransformer which translates model into API tests in Java. */
+public class JavaGapicSurfaceTestTransformer implements ModelToViewTransformer {
+
+  static String TEST_TEMPLATE_FILE = "java/test.snip";
+
+  private GapicCodePathMapper pathMapper;
+
+  public JavaGapicSurfaceTestTransformer(GapicCodePathMapper javaPathMapper) {
+    this.pathMapper = javaPathMapper;
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> views = new ArrayList<>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      SurfaceTransformerContext context =
+          SurfaceTransformerContext.create(
+              service, apiConfig, createTypeTable(), new JavaSurfaceNamer());
+      addImports(context);
+      GapicSurfaceTestClassView testClass = createTestClassView(service, context, apiConfig);
+      views.add(testClass);
+    }
+    return views;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    List<String> fileNames = new ArrayList<>();
+    fileNames.add(TEST_TEMPLATE_FILE);
+    // TODO(shinfan): Add gRPC service mock template.
+    return fileNames;
+  }
+
+  private ModelTypeTable createTypeTable() {
+    return new ModelTypeTable(new JavaTypeTable(), new JavaModelTypeNameConverter());
+  }
+
+  private void addImports(SurfaceTransformerContext context) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    typeTable.saveNicknameFor("org.junit.After");
+    typeTable.saveNicknameFor("org.junit.AfterClass");
+    typeTable.saveNicknameFor("org.junit.Before");
+    typeTable.saveNicknameFor("org.junit.BeforeClass");
+    typeTable.saveNicknameFor("org.junit.Test");
+    typeTable.saveNicknameFor("java.io.IOException");
+    typeTable.saveNicknameFor("java.util.List");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.ValueGenerator");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.LocalServiceHelper");
+    typeTable.saveNicknameFor("com.google.protobuf.GeneratedMessage");
+    typeTable.saveNicknameFor("junit.framework.Assert");
+  }
+
+  private GapicSurfaceTestClassView createTestClassView(
+      Interface service, SurfaceTransformerContext context, ApiConfig apiConfig) {
```

Comment:
Don't pass service or apiConfig as separate parameters, they are already passed in the context